File: SocialNetwork.Domain/Entity/System/Report.cs


using SocialNetwork.Domain.Abstractions;
using SocialNetwork.Domain.Entity.PostInfo;

namespace SocialNetwork.Domain.Entity.System
{
    public class Report : BaseAuditableEntity
    {
        public string Reason { get; set; }
        public string ReportType { get; set; }
        public string ResolutionNotes { get; set; }
        public string Status { get; set; }

        public string ReporterId { get; set; }
        public string? TargetUserId { get; set; }
        public Guid? TargetPostId { get; set; }
        public Guid? TargetCommentId { get; set; }

        public User Reporter { get; set; }
        public User? TargetUser { get; set; }
        public Post? TargetPost { get; set; }
        public Comment? TargetComment { get; set; }
        public DateTimeOffset? ResolvedAt { get; set; }

    }
}

File: SocialNetwork.WebApp/src/types/report.ts
import { CommentResource } from "./comment";
import { GroupResource } from "./group";
import { PostResource } from "./post";
import { UserResource } from "./user";

export type ReportResource = {
    id: string;
    reason: string;
    reportType: string;
    resolutionNotes: string;
    status: string;
  
    group: GroupResource;
    reporter: UserResource;
    targetUser: UserResource;
    targetPost: PostResource;
    targetGroup: GroupResource;
    targetComment: CommentResource;
    resolvedAt: Date;
    dateCreatedAt: Date;
  }
  